### PR TITLE
added-workspace

### PR DESCRIPTION
**Pull request by arpita-20**

### CODE REVIEW

File: prisma/migrations/20220930113030_altering_not_null_values_for_workspace_user_workspace_model/migration.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+  Warnings:
+
+  - Made the column `updatedAt` on table `UserWorkspace` required. This step will fail if there are existing NULL values in that column.
+  - Made the column `updatedAt` on table `Workspace` required. This step will fail if there are existing NULL values in that column.
+
+*/
+-- AlterTable
+ALTER TABLE "UserWorkspace" ALTER COLUMN "updatedAt" SET NOT NULL;
```
Comment (richa-copods) [inline prisma/migrations/20220930113030_altering_not_null_values_for_workspace_user_workspace_model/migration.sql:9]: Why we have separate migration for this?

File: prisma/migrations/20220930121320_adding_query/migration.sql [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+DECLARE 
+  nameVar varchar='aabcd1234';
+  userIdVar varchar;
+  firstNameVar varchar;
+  roleIDVar varchar;
+  random_id varchar;
+  userWorkspaceId varchar;
+  userList record;
+BEGIN
+    -- UPDATE "User" SET "firstName"=nameVar where email='careers@copods.co';
```
Comment (richa-copods) [inline prisma/migrations/20220930121320_adding_query/migration.sql:34]: Please remove commented code....for entire file

File: prisma/schema.prisma [hunk 3]
```diff
@@ -7,21 +7,50 @@ generator client {
   provider = "prisma-client-js"
 }
 
+model Workspace {
+  id            String          @id @default(cuid())
+  name          String
+  createdBy     User            @relation(fields: [createdById], references: [id])
+  createdById   String
+  UserWorkspace UserWorkspace[]
```
Comment (richa-copods) [inline prisma/schema.prisma:15]: Please maintain the consistency in naming fields, `UserWorkspace` U should be small

File: prisma/schema.prisma [hunk 4]
```diff
@@ -7,21 +7,50 @@ generator client {
...
+  createdAt     DateTime        @default(now())
+  updatedAt     DateTime        @updatedAt
+
+  @@unique([name, createdById])
+}
+
+model UserWorkspace {
+  id          String    @id @default(cuid())
+  Workspace   Workspace @relation(fields: [workspaceId], references: [id])
```
Comment (richa-copods) [inline prisma/schema.prisma:24]: Please maintain the consistency in naming fields, `Workspace` W should be small

File: prisma/migrations/20220930121320_adding_query/migration.sql [hunk 5]
```diff
@@ -0,0 +1,54 @@
+-- This is an empty migration.
+-- This is an empty migration.
+-- This is an empty migration.
+-- This is an empty migration.
+-- This is an empty migration.
+Create or replace function random_string(length integer) returns text as
```
Comment (richa-copods) [inline prisma/migrations/20220930121320_adding_query/migration.sql:6]: As we are always going to have length 25 and this function is not going to be reused, we can make this non-parametrised function and get rid of length checks at line 13 and so...

**Review by richa-copods — CHANGES_REQUESTED** [review]

File: prisma/schema.prisma (hunk 3 above)
Comment (arpita-20) [reply prisma/schema.prisma:15]: while making reference of models in different relation, first letter is capital plz check in "User" model also

File: prisma/migrations/20221003060144_adding_queries_for_workspace_creation/migration.sql [hunk 6]
```diff
@@ -0,0 +1,42 @@
+Create or replace function random_string() returns text as
```
Comment (arshabh-copods) [inline prisma/migrations/20221003060144_adding_queries_for_workspace_creation/migration.sql:1]: Please kindly explain - why we cannot go ahead with DB generated IDs and want to write our own function to generate IDs ?

I would like to see this function and this method to go away. In code also - we have at some place done stuff like this. Not a good thing

**Review by arshabh-copods — CHANGES_REQUESTED** [review]

**Pull request closed without merging**